File: FoostonWeb/ClientApp/components/ratings/Standing.ts
export class Standing {
    timeStamp: Date;
    rank: number;
    name : string;
    rating: string;
    gamesBack: string;
    previousRank: string ; 
    previousPoints: string ;
    weeksOff: string;
    change: string ;
    plusMinus: string ;
    wins: string;
    losses: string;
    percent: string;

    constructor(csv: string, timeStamp: Date) {
        this.timeStamp = timeStamp;
        const fields = csv.split("\t");
        this.rank = Number(fields[0]);
        this.name = fields[1];
        this.rating = fields[2];
        this.gamesBack = fields[3];
        this.previousRank = fields[4];
        this.previousPoints =fields[5];
        this.weeksOff = fields[6];
        this.change =fields[7];
        this.plusMinus =fields[8];
        this.wins =fields[9];
        this.losses =fields[10];
        this.percent =fields[11];
    }
}

export interface RatingsViewModel {
    currentStandings: Standing[];
    otherDates: Date[];
}

File: FoostonWeb/Models/Standing.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace FoostonWeb.Models {
    public class Standing {
        public int Id { get; set; }
        public DateTime TimeStamp { get; set; }
        public int Rank { get; set; }
        public string Name { get; set; }
        public string Rating { get; set; }
        public string GamesBack { get; set; }
        public string PreviousRank { get; set; }
        public string PreviousPoints { get; set; }
        public string WeeksOff { get; set; }
        public string Change { get; set; }
        public string PlusMinus { get; set; }
        public string Wins { get; set; }
        public string Losses { get; set; }
        public string Percent { get; set; }
    }
}